internal/test/framework/framework.go

```diff
@@ -29,6 +31,34 @@ type TestFramework struct {
 	K8sclientset *kubernetes.Clientset
 	// OSConfigClient is the OpenShift config client, we will use to query the OpenShift api object status
 	OSConfigClient *configclient.Clientset
+	// noTeardown is an indicator that the user supplied the VMs and they should not be destroyed
+	noTeardown bool
+}
+
+// Creds is used for parsing the vmCreds command line argument
+type Creds []*types.Credentials
+
+// Set populates the list of credentials from the vmCreds command line argument
+func (c *Creds) Set(value string) error {
+	if value == "" {
+		return nil
+	}
+
+	splitValue := strings.Split(value, ",")
+	// Credentials consists of three elements, so this has to be
+	if len(splitValue)%3 != 0 {
+		return fmt.Errorf("incomplete VM credentials provided")
+	}
+
+	for i := 0; i < len(splitValue); i += 3 {
+		cred := types.NewCredentials(splitValue[i], splitValue[i+1], splitValue[i+2])
```

Comment:
Have some validation around the args being passed as well. Even if you are working on 3 items at a time, you cannot assume that order is preserved for `instancedID`, `password` etc. Perhaps a TODO?